packit/patches.py

```diff
@@ -231,6 +235,77 @@ def linearize_history(git_ref: str):
             env={"FILTER_BRANCH_SQUELCH_WARNING": "1"},
         )
 
+    @staticmethod
+    def process_patches(
+        patches: Dict[str, str], commits: List[git.Commit], patch_list: List
+    ):
+        """ process patches (created with git-format) """
+        for commit in commits:
+            for patch_name, patch_content in patches.items():
+                # `git format-patch` usually creates one patch for a merge commit,
+                # so some commits won't be covered by a dedicated patch file
+                if commit.hexsha in patch_content:
+                    path = Path(patch_name)
+                    patch_metadata = PatchMetadata.from_commit(
+                        commit=commit, patch_path=path
+                    )
+
+                    if patch_metadata.ignore:
+                        logger.debug(
+                            f"[IGNORED: {patch_metadata.name}] {commit.summary}"
+                        )
+                    else:
+                        logger.debug(f"[{patch_metadata.name}] {commit.summary}")
+                        patch_list.append(patch_metadata)
+                    break
+
+    @staticmethod
+    def process_git_am_style_patches(patch_list: List[PatchMetadata],):
+        """
+        When using `%autosetup -S git_am`, there is a case
+        where a single patch file contains multiple commits.
+        This is problematic for us since only the HEAD patch commit
+        is annotated.
+
+        In this case, we need to:
+        1. detect this is happening (there are some annotated commits)
+        2. ...
+        """
+        # are there some patches which are present in spec and some which are not?
+        some_patches_not_in_spec = any(
+            [not bool(p.present_in_specfile) for p in patch_list]
+        ) and any([bool(p.present_in_specfile) for p in patch_list])
```

Comment:
Using generators might speed things up in case of many patches
```suggestion
            (not bool(p.present_in_specfile) for p in patch_list)
        ) and any((bool(p.present_in_specfile) for p in patch_list))
```

Reply:
`present_in_specfile` is already defined as `bool`, so
```suggestion
            not p.present_in_specfile for p in patch_list
        ) and any(p.present_in_specfile for p in patch_list)
```

Reply:
I dropped this code from the new implementation